marc_export/src/main/java/whelk/export/marc/ProfileExport.java

```diff
@@ -137,38 +138,26 @@ else if (collection.equals("hold") && updateShouldBeExported(id, collection, pro
         {
             List<Document> versions = m_whelk.getStorage().loadAllVersions(id);
 
-            // Export the new/current itemOf
-            Document version;
-            if (versions.size() > 0)
+            for (Document version : versions)
             {
-                version = versions.get(0);
-            }
-            else
-            {
-                version = m_whelk.getStorage().load(id);
-            }
-            String itemOf = version.getHoldingFor();
-            String itemOfSystemId = m_whelk.getStorage().getSystemIdByIri(itemOf);
-            if (itemOfSystemId != null) // itemOfSystemId _can_ be null, if the bib linked record is deleted (no thing-uri left in the id table)
-                exportDocument(
-                        m_whelk.getStorage().loadEmbellished(itemOfSystemId, m_whelk.getJsonld())
-                        , profile, output, exportedIDs, deleteMode, doVirtualDeletions);
+                String itemOf = version.getHoldingFor();
+                Instant modified = ZonedDateTime.parse(version.getModified()).toInstant();
 
-            // If the itemOf link was changed, also export the bib that is no longer linked.
-            if (versions.size() > 1)
-            {
-                Document oldVersion = versions.get(1);
-                String oldItemOf = oldVersion.getHoldingFor();
-                if (!oldItemOf.equals(itemOf))
-                {
-                    String oldItemOfSystemId = m_whelk.getStorage().getSystemIdByIri(oldItemOf);
-                    if (oldItemOfSystemId != null) // oldItemOfSystemId _can_ be null, if the bib linked record is deleted (no thing-uri left in the id table)
-                        exportDocument(
-                                m_whelk.getStorage().loadEmbellished(oldItemOfSystemId, m_whelk.getJsonld())
-                                , profile, output, exportedIDs, deleteMode, doVirtualDeletions);
-                }
+                if (modified.isAfter(until.toInstant()))
+                    continue;
+
+                String itemOfSystemId = m_whelk.getStorage().getSystemIdByIri(itemOf);
+                if (itemOfSystemId != null) // itemOfSystemId _can_ be null, if the bib linked record is deleted (no thing-uri left in the id table)
+                    exportDocument(
+                            m_whelk.getStorage().loadEmbellished(itemOfSystemId, m_whelk.getJsonld())
+                            , profile, output, exportedIDs, deleteMode, doVirtualDeletions);
+
+                boolean insideInterval = from.toInstant().isBefore(modified) && until.toInstant().isAfter(modified);
+                if ( !insideInterval )
+                    break; // Only one more once outside the interval, then stop.
```

Comment:
These three lines and the `versions` loop are kinda difficult to grasp, unless you know the ordering of the versions. Can you clarify the comment and explain why the `insideInterval` makes sense here instead of e.g. at the top of the loop?